raiden/connection_manager.py

```diff
@@ -109,14 +111,29 @@ def connect(
                     funding
                 )
 
-    def leave(self, wait_for_settle=True, timeout=30):
+    def leave_async(self, wait_for_settle=True, timeout=240):
+        """ Async version of `leave()`
+        Args:
+            wait_for_settle (bool): block until successful settlement?
+            timeout (float): maximum time to wait in seconds
         """
-        Leave the token network.
+        leave_result = AsyncResult()
+        gevent.spawn(
+            self.leave,
+            wait_for_settle=wait_for_settle,
+            timeout=timeout,
+            async_result=leave_result
+        )
+        return leave_result
+
+    def leave(self, wait_for_settle=True, timeout=30, async_result=AsyncResult()):
```

Comment:
`timeout` is a bit weird in the `async` scenario, IMO the correct way is to implement the `async` version first and the the synchronous calls the `async` and use the `event.wait` with a `timeout`, this way we don't need to invent our own timeout handling and there is not overlap of what was specified in the `leave` call and the `async.wait` call